qemu/tests/max_channel_lun.py

```diff
@@ -0,0 +1,84 @@
+import logging
+import re
+import random
+
+from virttest import error_context
+from virttest import env_process
+from virttest import virt_vm
+from virttest import qemu_storage
+from virttest import data_dir
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Verify max luns support for one channel of spapr-vscsi.
+
+    Step:
+     1. Boot a guest with 32 luns for one channel
+     2. Boot a guest with 33 luns for one channel
+
+    :param test: Qemu test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+
+    vm = env.get_vm(params['main_vm'])
+    timeout = float(params.get("login_timeout", 240))
+    stg_image_num = int(params.get('stg_image_num'))
+    stg_image_name = params.get('stg_image_name', 'images/%s')
+    channel = params.get('channel')
+    error_info = params["error_info"]
+
+    for i in range(stg_image_num):
+        name = "stg%d" % i
+        params['images'] += " %s" % name
+        params["image_name_%s" % name] = stg_image_name % name
+        params["blk_extra_params_%s" % name] = channel
+        params["drive_port_%s" % name] = i
+        params["scsi_hba_%s" % name] = "spapr-vscsi"
+    if params['luns'] == "lun_33":
+        img_params = params.object_params("stg32")
+        image = qemu_storage.QemuImg(img_params, data_dir.get_data_dir(),
+                                     "stg32")
+        params["extra_params"] = "-blockdev node-name=file_stg32,\
```

Comment:
`blockdev` is only supported since qemu >= 3.0, and this is a negative test, so it’s up to you whether to consider the `-drive` mode. If only `blockdev` requires, please add `required_qemu = [3.0, )`, thanks.

Reply:
32 lun for one channel is supported from qemu3.1,so keep using blockdev.